src/Dotnet.Script.Core/ScriptCompiler.cs

```diff
@@ -45,15 +47,15 @@ public ScriptCompiler(ScriptLogger logger)
 
         public virtual ScriptOptions CreateScriptOptions(ScriptContext context)
         {
-            var opts = ScriptOptions.Default.
-                AddImports(ImportedNamespaces).
-                AddReferences(ReferencedAssemblies).
-                WithSourceResolver(SourceFileResolver.Default).
-                WithMetadataResolver(ScriptMetadataResolver.Default);
+            var opts = ScriptOptions.Default.AddImports(ImportedNamespaces)
+                .AddReferences(ReferencedAssemblies)
+                .WithSourceResolver(SourceFileResolver.Default)
+                .WithMetadataResolver(ScriptMetadataResolver.Default)
+                .WithEmitDebugInformation(context.DebugMode);            
 
             if (!string.IsNullOrWhiteSpace(context.FilePath))
             {
-                opts = opts.WithFilePath(context.FilePath);
+                opts = opts.WithFilePath(context.FilePath).WithFileEncoding(Encoding.UTF8);
```

Comment:
same here, encoding will not be needed if you change here https://github.com/filipw/dotnet-script/blob/master/src/Dotnet.Script.Core/ScriptCompiler.cs#L108 to use the new stream-based API https://github.com/dotnet/roslyn/blob/master/src/Scripting/CSharp/CSharpScript.cs#L44